This... is interesting. Roughly what I expected, actually (I expected some sort of assembly or synchronization primitive to go wrong).

```
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/runtime_cortexm_qemu.go:70:10: too many runtime instructions
  call void asm sideeffect "wfi", ""(), !dbg !596

traceback:
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/runtime_cortexm_qemu.go:70:10:
  call void asm sideeffect "wfi", ""(), !dbg !596
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/runtime_cortexm_qemu.go:57:6:
  call void @runtime.exit(i32 1, i8* undef), !dbg !588
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/panic.go:20:7:
  call void @runtime.abort(i8* undef), !dbg !596
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/panic.go:32:14:
  call void @runtime.runtimePanic(i8* getelementptr inbounds ([23 x i8], [23 x i8]* @"runtime$string.69", i32 0, i32 0), i32 23, i8* undef), !dbg !589
/home/ayke/src/github.com/tinygo-org/tinygo/src/internal/task/task_stack.go:43:18:
  call void @runtime.nilPanic(i8* undef), !dbg !591
/home/ayke/src/github.com/tinygo-org/tinygo/src/runtime/scheduler_any.go:16:12:
  call void @"internal/task.Pause"(i8* undef), !dbg !600
/home/ayke/src/github.com/tinygo-org/tinygo/testdata:
  call void @time.Sleep(i64 1000000, i8* undef), !dbg !589
```

~I guess I'll just mark inline assembly like the `br` instruction with unknown operands.~

--- a/interp/interpreter.go
+++ b/interp/interpreter.go
@@ -219,6 +219,9 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 				if err != nil {
 					return nil, mem, err
 				}
+			case callFn.name == "internal/task.Pause":
+				// Task scheduling isn't possible at compile time.
+				return nil, mem, r.errorAt(inst, errUnsupportedRuntimeInst)
 			case callFn.name == "runtime.nanotime" && r.pkgName == "time":
 				// The time package contains a call to runtime.nanotime.
 				// This appears to be to work around a limitation in Windows

--- a/interp/interpreter.go
+++ b/interp/interpreter.go
@@ -81,12 +81,28 @@ func (r *runner) run(fn *function, params []value, parentMem *memoryView, indent
 		if inst.opcode != llvm.PHI {
 			for _, v := range inst.operands {
 				if v, ok := v.(localValue); ok {
-					if localVal := locals[fn.locals[v.value]]; localVal == nil {
+					index, ok := fn.locals[v.value]
+					if !ok {
+						// This is a localValue that is not local to the
+						// function. An example would be an inline assembly call
+						// operand.
+						isRuntimeInst = true
+						break
+					}
+					localVal := locals[index]
+					if localVal == nil {
+						// Trying to read a function-local value before it is
+						// set.
 						return nil, mem, r.errorAt(inst, errors.New("interp: local not defined"))
 					} else {
 						operands = append(operands, localVal)
 						if _, ok := localVal.(localValue); ok {
+							// The function-local value is still just a
+							// localValue (which can't be interpreted at compile
+							// time). Not sure whether this ever happens in
+							// practice.
 							isRuntimeInst = true
+							break
 						}
 						continue
 					}

